twofactor_privacyidea/lib/Provider/TwoFactorPrivacyIDEAProvider.php

```diff
@@ -279,45 +299,50 @@ public function authenticate($username, $password) {
         $transaction_id = $this->request->getParam("transaction_id");
         $signatureData = $this->request->getParam("signatureData");
         $clientData = $this->request->getParam("clientData");
-        $this->logger->debug("transaction_id: " . $transaction_id);
-        $this->logger->debug("signatureData: " . $signatureData);
-        $this->logger->debug("clientData: " . $clientData);
+        $this->log("debug", "transaction_id: " . $transaction_id);
+        $this->log("debug", "signatureData: " . $signatureData);
+        $this->log("debug", "clientData: " . $clientData);
 
         if ($transaction_id) {
             // add transaction ID in case of challenge response
             $options['body']["transaction_id"] = $transaction_id;
         }
 
         if ($signatureData) {
-            $this->logger->debug('We are doing a U2F response.');
+            $this->log("debug", "We are doing a U2F response.");
             // here we add the signatureData and the clientData in case of U2F
             $options['body']["signaturedata"] = $signatureData;
             $options['body']["clientdata"] = $clientData;
         }
 
         $errorCode = 0;
+        $res = 0;
         try {
             $client = $this->httpClientService->newClient();
             $res = $client->post($url, $options);
-            if ($res->getStatusCode() === 200) {
-                $body = $res->getBody();
-                $body = json_decode($body);
-                if ($body->result->status === true) {
-                    if ($body->result->value === true) {
-                        return true;
-                    } else {
-                        $error_message = $this->trans->t("Failed to authenticate.");
-                        $errorCode = 1;
-                    }
+	        $body = $res->getBody();
+	        $body = json_decode($body);
+
+            if ($body->result->status === true) {
+                if ($body->result->value === true) {
+                    return true;
                 } else {
-                    $error_message = $this->trans->t("Failed to authenticate. privacyIDEA error.");
+                    $error_message = $this->trans->t("Failed to authenticate.");
+                    $errorCode = 1;
+                    $this->log("info", "User failed to authenticate. Wrong OTP value.");
```

Comment:
This could be wrong otp value or wrong username?
I would simply call it "user failed to authenticate".

Reply:
Should we maybe add the *username* to the log entry? After all it is a log file, so the username would be ok. And this would help to see, if there is a wrong username used.

~~~~php
$this->log("info", "User {$username}s failed to authenticate.");
~~~~

Reply:
It can't be wrong username. After logging in with username and password, the user needs to enter the OTP. If the user needs OTP, the username and the password are definitely correct. So this message appears, when the OTP is incorrect.

We do not need to add the username, because for every log, you get in ownCloud, the username is listed like this
`"user":"username","app":"privacyIDEA","message":"User failed to authenticate. Wrong OTP value."`

Reply:
OK. Thanks.